server.go

```diff
@@ -116,9 +161,15 @@ func (server *Server) Stop() (err error) {
 }
 
 // Public interface to get access to server messages
-// Returns slice of message pointers
-func (server *Server) Messages() []*Message {
-	return server.messages.items
+// Returns copy of message data
+func (server *Server) Messages() []Message {
+	server.Lock()
+	defer server.Unlock()
+	copiedItems := []Message{}
+	for i := 0; i < len(server.messages.items); i++ {
```

Comment:
Should we have thread safe method `messages#items()` for reading this slice of pointers:

```go
type messages struct {
	sync.RWMutex
	messages []*Message
}

func (messages *messages) append(item *Message) {
	messages.Lock()
	defer messages.Unlock()

	messages.messages = append(messages.messages, item)
}

func (messages *messages) items() []*Message {
	messages.Lock()
	defer messages.Unlock()

	return messages.items
}
```

Reply:
Yes, that seems sensible to do that. I appreciate your understanding.

Okay, shall I quickly code up these additional bits? 

1. This `items()` method on `*messages` exactly as you have detailed above.
2. Changing all the methods on `*Message` to being methods on `Message` - this makes additional sense given the change to return `[]Message` as we have discussed.

Reply:
1. Yes.
2. Of course yes (as expected). We need change all methods from `func (message *Message)` to `func (message Message)`

Reply:
Hi Vladislav, actually, when I went to add this `items()` method, I looked it where it is being used... it's only in the tests and in all cases the `Messages()` method on `Server` type can (and should) be used instead.

Reply:
@benjamin-rood I prefer to use public interfaces, I found just in one test file.

<img width="447" alt="Screenshot 2022-11-13 at 21 48 23" src="https://user-images.githubusercontent.com/14002692/201543880-49cd81f3-642b-4e2c-ae24-bed84c86ff13.png">

As I understend we no need to implement messages.items getter with lock, am I right?